src/temaWeek8Exception/main/Repository.java

```diff
@@ -0,0 +1,101 @@
+package temaWeek8Exception.main;
+
+import java.time.Year;
+import java.util.*;
+
+public class Repository {
+	String firstName = "";
+	String lastName = "";
+	String dateOfBirth = "";
+	String gender = "";
+	String ID = "";
+	
+	
+	Student student = new Student(firstName, lastName, dateOfBirth, gender, ID);
+	List<Student> studentRepo = new ArrayList<>();
+	
+	public static void main(String[] args) {
+	
+	}
+	
+//	create student method
+	public Student createStudent(String firstName, String lastName, String dateOfBirth, String gender, String ID) {
+
+		firstName = student.addFirstName(firstName);//calling addFirstName method for validation of method argument
+		lastName = student.addLastName(lastName);//calling addLastName method for validation of method argument
+		dateOfBirth = student.addDateOfBirth(dateOfBirth);//calling addDateOfBirth method for validation of method argument
+		gender = student.addGender(gender);//calling addGender method for validation of method argument
+		ID = student.addID(ID);//calling addID method for validation of method argument
+		Student s = new Student(firstName, lastName, dateOfBirth, gender, ID);
+		
+		return s;
+	}
+	
+	//	method for retrieving students of same  age and creating a new list called sameAge list
+	public List<Student> retrieveStudentWithSameAge(String ageX,List<Student> studentRepo) throws Student.ExtendedException {
```

Comment:
There is a list which holds the students as a member of the class. On the other hand, this method passes a list as parameter.
The design should be improved. I'm for the second variant, where you have this list passed to the method. If you choose to do so, the `studentRepo` member of the class should be removed.